src/PaymentPart/Output/HtmlOutput/HtmlOutput.php

```diff
@@ -0,0 +1,182 @@
+
+        $paymentPart = str_replace('{{ amount-content }}', $amountContent, $paymentPart);
+
+        return $paymentPart;
+    }
+
+    private function addAmountContentReceipt(string $paymentPart): string
+    {
+        $amountContent = '';
+
+        foreach($this->getAmountElementsReceipt() as $amountElement) {
+            $amountContent .= $this->getContentElement($amountElement);
+        }
+
+        $paymentPart = str_replace('{{ amount-content-receipt }}', $amountContent, $paymentPart);
+
+        return $paymentPart;
+    }
+
+    private function addFurtherInformationContent(string $paymentPart): string
+    {
+        $furtherInformationContent = '';
+
+        foreach($this->getFurtherInformationElements() as $furtherInformationElement) {
+            $furtherInformationContent .= $this->getContentElement($furtherInformationElement);
+        }
+
+        $paymentPart = str_replace('{{ further-information-content }}', $furtherInformationContent, $paymentPart);
+
+        return $paymentPart;
+    }
+
+    private function addPrintableContent(string $paymentPart): string
+    {
+        $printableStyles = '';
+        if ($this->isPrintable()) {
+            $printableStyles = PrintableStylesTemplate::TEMPLATE;
+        }
+
+        $paymentPart = str_replace('{{ printable-content }}', $printableStyles, $paymentPart);
+
+        return $paymentPart;
+    }
+
+    private function getContentElement(OutputElementInterface $element): string
+    {
+        if ($element instanceof Title) {
+            $elementTemplate = TitleElementTemplate::TEMPLATE;
+            $elementString = str_replace('{{ title }}', $element->getTitle(), $elementTemplate);
+
+            return $elementString;
+        }
+
+        if ($element instanceof Text) {
+            $elementTemplate = TextElementTemplate::TEMPLATE;
+            $elementString = str_replace('{{ text }}', nl2br($element->getText()), $elementTemplate);
+
+            return $elementString;
+        }
+
+        if ($element instanceof Placeholder) {
+            $elementTemplate = PlaceholderElementTemplate::TEMPLATE;
+            $elementString = $elementTemplate;
+
+            $svgDoc = new \DOMDocument();
+            $svgDoc->loadXML(file_get_contents($element->getFile()));
+            $svg = $svgDoc->getElementsByTagName('svg');
+            $dataUri = 'data:image/svg+xml;base64,' . base64_encode($svg->item(0)->C14N());
+
+            $elementString = str_replace('{{ file }}', $dataUri, $elementString);
+            $elementString = str_replace('{{ width }}', $element->getWidth(), $elementString);
+            $elementString = str_replace('{{ height }}', $element->getHeight(), $elementString);
+            $elementString = str_replace('{{ id }}', $element->getType(), $elementString);
+
+            return $elementString;
+        }
+    }
+
+    private function translateContents($paymentPart, $language)
+    {
+        $translations = Translation::getAllByLanguage($language);
```

Comment:
Maybe it could be used as a "translator" service injected in class that builds this object?